### PR TITLE
Allow modifying webroot-map during renewal

### LINKED ISSUE #2977: --dry-run on renew didn't work
Hello,

I just try to test the renew like this:
./letsencrypt-auto renew -v --dry-run
and the log says:

```
2016-05-12 21:03:20,463:DEBUG:certbot.main:Root logging level set at 20
2016-05-12 21:03:20,464:INFO:certbot.main:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2016-05-12 21:03:20,464:DEBUG:certbot.main:certbot version: 0.6.0
2016-05-12 21:03:20,464:DEBUG:certbot.main:Arguments: ['-v', '--dry-run']
2016-05-12 21:03:20,464:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#apache,PluginEntryPoint#webroot,PluginEntryPoint#null,PluginEntryPoint#manual,PluginEntryPoint#standalone)
2016-05-12 21:03:20,465:DEBUG:certbot.main:Exiting abnormally:
Traceback (most recent call last):
  File "/root/.local/share/letsencrypt/bin/letsencrypt", line 11, in <module>
    sys.exit(main())
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 693, in main
    return config.func(config, plugins)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/main.py", line 538, in renew
    renewal.renew_all_lineages(config)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/certbot/renewal.py", line 305, in renew_all_lineages
    raise errors.Error("Currently, the renew verb is only capable of "
Error: Currently, the renew verb is only capable of renewing all installed certificates that are due to be renewed; individual domains cannot be specified with this action. If you would like to renew specific certificates, use the certonly command. The renew verb may provide other options for selecting certificates to renew in the future.
```

I use a cli.ini with 
authenticator = webroot
webroot-map = {....some domains which are correctly installed with certonly command }

Any idea?

Thanks, Alex

### PR DESCRIPTION
**Pull request by bmw**
Fixes #2977.

### CODE REVIEW

File: certbot/renewal.py [hunk 1]
```diff
@@ -301,7 +301,10 @@ def _renew_describe_results(config, renew_successes, renew_failures,
 def renew_all_lineages(config):
     """Examine each lineage; renew if due and report results"""
 
-    if config.domains != []:
+    # This is trivially False if config.domains is empty
+    if any(domain not in config.webroot_map for domain in config.domains):
```
Comment (pde) [inline certbot/renewal.py]: This is probably correct, though we should convince ourselves that there are no other ways that `config.domains` might get set, aside from the webroot plugin.  For instance, if there's a cli.ini file with `domains` set to we want to be erroring out here?

**Comment by pde** [comment]
If we're sure that that's the best approach for now (as opposed to allowing any `domains` setting method as long as it matches) then LGTM.

**Comment by pde** [comment]
Ooops sorry caught the wrong username in a dropdown menu :)

File: certbot/renewal.py (hunk 1 above)
Comment (bmw) [inline certbot/renewal.py]: I think so. This is what we have been doing in the past and in general we cannot differentiate between values set on the command line and those in a CLI config file.

**Comment by bmw** [comment]
> as opposed to allowing any domains setting method as long as it matches

Do you mean using domains as a selector for choosing which lineages to renew? I personally don't think we should do this right now. Cert/lineage management has proven to be quite difficult and I think we should think carefully about how we want to implement it (James' cert manager?) before doing anything like this. Otherwise, we we'll have multiple ways to specify the same thing, which may conflict with each other in the future.

**Comment by pde** [comment]
@bmw no I just mean being tolerant if someone has `domains = blah` in their `cli.ini` file, and the domains in that file match the certs being renewed.

**Comment by pde** [comment]
So basically, don't error if `domains` exactly matches the set of domains being renewed.

Intuitively I think that would be polite behaviour, but it might be a little tricky to implement.

For several reasons I've been thinking that the `renew.renew_all_lineages` might need to be refactored to walk all the renewal conf files before commencing any actual renewal work; this might be another reason to do that.

**Comment by bmw** [comment]
Ah I understand. I think adding something like that would be fine, however, it'll require significant refactoring and should be thoroughly tested given the importance of `renew`. Do you think something like this needs to be in the 0.7.0 release?

I personally would prefer not to block this PR on this feature though. Perhaps we should just open a new issue?

**Comment by pde** [comment]
I got a report of that sort of breakage (perhaps from Seth?) yesterday afternoon, actually.  But yes let's make it a different ticket.

**Pull request merged**